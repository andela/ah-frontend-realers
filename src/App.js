import React, { Component } from "react";

class App extends Component {
  render() {
    return (
        <h1> Welcome to Author's Haven</h1>
    );
  }
}
export default App;