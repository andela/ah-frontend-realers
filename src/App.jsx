/* eslint-disable import/no-named-as-default */
import React from 'react';
import './App.scss';
import { Provider } from 'react-redux';
import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';
import { ToastContainer } from 'react-toastify';
import HeaderSection from './commons/components/HeaderSection';
import FooterSection from './commons/components/FooterSection';
import Landing from './Landing/containers/Landing';
import LoginContainer from './auth/Login/containers/LoginContainer';
import Search from './Articles/containers/SearchArticles';
import ViewArticleContainer from './ViewArticle/container/ViewArticleContainer';
import 'react-toastify/dist/ReactToastify.css';
import SignUp from './auth/signup/components/SignUp';
import UserProfile from './Profiles/components/UserProfile';
import editProfile from './Profiles/components/EditProfile';
import AllArticles from './Articles/components/AllArticles';
import ResetPasswordContainer from './auth/passwordreset/components/ResetComponent';
import store from './Mainstore/Store';
import CreateArticle from './Articles/containers/CreateArticle';
import history from './commons/history';

export default function App() {
  return (
    <Provider store={store}>
      <Router history={history}>
        <HeaderSection />
        <ToastContainer />
        <Route path="/login" exact component={LoginContainer} />
        <Switch>
          <Route path="/" exact component={Landing} />
          <Route path="/about" component={Landing} />
          <Route path="/signup" component={SignUp} />
          <Route path="/articles" component={AllArticles} />
          <Route path="/reset" component={ResetPasswordContainer} />
          <Route path="/search/:searchkey" exact component={Search} />
          <Route path="/create-article" component={CreateArticle} />
          <Route path="/article/:slug" component={ViewArticleContainer} />
          <Route path="/profile" component={UserProfile} />
          <Route path="/editprofile" component={editProfile} />
        </Switch>
        <FooterSection />
      </Router>
    </Provider>
  );
}
