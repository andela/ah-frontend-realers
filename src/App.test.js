import React from 'react';
import { shallow } from 'enzyme';
import App from './App';

// tests for this
describe('App Component', () => {
  it('should match the snapshot', () => {
    const component = shallow(<App />);
  
    expect(component).toMatchSnapshot();
  });
});