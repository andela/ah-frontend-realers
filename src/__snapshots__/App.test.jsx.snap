// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component should match the snapshot 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <BrowserRouter
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <HeaderSection />
    <ToastContainer
      autoClose={5000}
      bodyClassName={null}
      className={null}
      closeButton={
        <CloseButton
          ariaLabel="close"
        />
      }
      closeOnClick={true}
      draggable={true}
      draggablePercent={80}
      hideProgressBar={false}
      newestOnTop={false}
      pauseOnFocusLoss={true}
      pauseOnHover={true}
      position="top-right"
      progressClassName={null}
      progressStyle={null}
      role="alert"
      rtl={false}
      style={null}
      toastClassName={null}
      transition={[Function]}
    />
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(LoginContainer)",
          "type": [Function],
        }
      }
      exact={true}
      path="/login"
    />
    <Switch>
      <Route
        component={
          Object {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "displayName": "Connect(Landing)",
            "type": [Function],
          }
        }
        exact={true}
        path="/"
      />
      <Route
        component={
          Object {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "displayName": "Connect(Landing)",
            "type": [Function],
          }
        }
        path="/about"
      />
      <Route
        component={
          Object {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "displayName": "Connect(SignUpUserForm)",
            "type": [Function],
          }
        }
        path="/signup"
      />
      <Route
        component={
          Object {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "displayName": "Connect(AllArticles)",
            "type": [Function],
          }
        }
        path="/articles"
      />
      <Route
        component={
          Object {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "displayName": "Connect(ResetPasswordContainer)",
            "type": [Function],
          }
        }
        path="/reset"
      />
      <Route
        component={
          Object {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "displayName": "Connect(SearchArticles)",
            "type": [Function],
          }
        }
        exact={true}
        path="/search/:searchkey"
      />
      <Route
        component={
          Object {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "displayName": "Connect(CreateArticle)",
            "protoTypes": Object {
              "isArticlePosted": [Function],
              "postAnArticle": [Function],
            },
            "type": [Function],
          }
        }
        path="/create-article"
      />
      <Route
        component={
          Object {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "displayName": "Connect(ArticleContainer)",
            "type": [Function],
          }
        }
        path="/article/:slug"
      />
      <Route
        component={[Function]}
        path="/profile"
      />
      <Route
        component={[Function]}
        path="/editprofile"
      />
    </Switch>
    <FootSection />
  </BrowserRouter>
</Provider>
`;
